walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,64 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
+
+  /**
+   * Returns collection of time slots, in which we can appoint the meeting.
+   * @param events   All events in the day.
+   * @param request  Information about event that we want to add (attendees, required duration).
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // hash map, where key is the moment of time and object is
+    // the change in the number of available attendees at that particular time
+    HashMap<Integer, Long> timePointsOfAvailabilityChange = new HashMap<>();
+    Collection<String> attendees = request.getAttendees();
+
+    long numberOfAttendeesRequired = attendees.size();
+    long requiredDuration = request.getDuration();
+
+    // add start and end times of the day
+    timePointsOfAvailabilityChange.put(0, numberOfAttendeesRequired);
+    timePointsOfAvailabilityChange.put(TimeRange.WHOLE_DAY.end(), -numberOfAttendeesRequired);
+
+    // add all time point, where the number of free attendees is changing
+    for (Event event : events) {
+      // the event can have attendees that are not present in the request,
+      // so we shouldn't count them
+      Collection<String> eventAttendees = new ArrayList<>(event.getAttendees());
```

Comment:
if I understood correctly, these are the attendees that are not on the request. So two things here:

- change the inline comment slightly to say "the event can _only_ have ...."
- change the variable name to something more precise, e.g. `eventAttendeesNotOnMeetingRequest`